docs/cpan-dependency-confusion.md

```diff
@@ -0,0 +1,239 @@
+# CPAN dependency confusion
+## Rationale 
+Take advantage of an incorrectly configured setup that resolves a public module (published on CPAN) in place of the expected private/local version due to one of the following reasons:
+- Local module is not installed at the time of running the CPAN client
+- Public CPAN version is preferred to local
+- Using a CPAN repository (proxy) that can be confused
+
+This type of vulnerability is presented in [Dependency Confusion](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610) by [Alex Birsan](https://medium.com/@alex.birsan)
+
+## Background
+"Dependency confusion" takes advantage of mixing (semi-)private modules with public CPAN modules.
+
+In the Perl ecosystem, it could be one or more of the following options:
+- Using a "DarkPAN" along with CPAN
+- Using a "GreyPAN" along with CPAN
```

Comment:
"GreyPAN" isn't a term I've heard before. I like it.

Would you mind adding your definition of it to the `glossary.md`?

Reply:
Added :)